packages/cozy-stack-client/src/OAuthClient.js

```diff
@@ -0,0 +1,358 @@
+    )
+      result['redirect_uris'] = [result['redirect_uris']]
+
+    return result
+  }
+
+  /**
+   * Converts a snake-cased data set to camel case, suitable for internal use
+   * @param   {object} data Initial data
+   * @returns {object} Formatted data
+   * @private
+   */
+  camelCaseOAuthData(data) {
+    const mappedFields = {
+      client_id: 'clientID',
+      client_name: 'clientName',
+      client_secret: 'clientSecret',
+      registration_access_token: 'registrationAccessToken',
+      software_id: 'softwareID'
+    }
+
+    const result = {}
+
+    Object.keys(data).forEach(fieldName => {
+      const key = mappedFields[fieldName] || fieldName
+      const value = data[fieldName]
+      result[key] = value
+    })
+
+    return result
+  }
+
+  /**
+   * Registers the currenly configured client with the OAuth server. 
+   * @throws {Error} When the client is already registered
+   * @returns {promise} A promise that resolves with a complete list of client information, including client ID and client secret.
+   */
+  async register() {
+    if (this.isRegistered()) throw new Error('Client already registered')
+
+    const data = await this.fetch(
+      'POST',
+      '/auth/register',
+      this.snakeCaseOAuthData({
+        redirectURI: this.oauthOptions.redirectURI,
+        clientName: this.oauthOptions.clientName,
+        softwareID: this.oauthOptions.softwareID,
+        clientKind: this.oauthOptions.clientKind,
+        clientURI: this.oauthOptions.clientURI,
+        logoURI: this.oauthOptions.logoURI,
+        policyURI: this.oauthOptions.policyURI,
+        softwareVersion: this.oauthOptions.softwareVersion,
+        notificationPlatform: this.oauthOptions.notificationPlatform,
+        notificationDeviceToken: this.oauthOptions.notificationDeviceToken
+      })
+    )
+    
+    this.oauthOptions = this.camelCaseOAuthData({
+      ...this.oauthOptions,
+      client_id: data.client_id,
+      client_name: data.client_name,
+      client_secret: data.client_secret,
+      registration_access_token: data.registration_access_token,
+      software_id: data.software_id
+    })
+
+    return this.oauthOptions
+  }
+
+  /**
+   * Unregisters the currenly configured client with the OAuth server. 
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @returns {promise}
+   */
+  async unregister() {
+    if (!this.isRegistered()) throw new NotRegisteredException()
+
+    return this.fetch(
+      'DELETE',
+      `/auth/register/${this.oauthOptions.clientID}`,
+      null,
+      {
+        credentials: this.registrationAccessTokenToAuthHeader()
+      }
+    )
+  }
+
+  /**
+   * Fetches the complete set of client informations from the server after it has been registered.
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @returns {promise}
+   */
+  async fetchInformations() {
```

Comment:
_"information"_ in english may always be written in a plural form and don't take a final "s".